src/test/java/org/opensearch/security/dlic/rest/api/UserApiTest.java

```diff
@@ -133,6 +133,10 @@ public void testUserApi() throws Exception {
         response = rh.executePutRequest(ENDPOINT + "/internalusers/", "{\"hash\": \"123\"}", new Header[0]);
         Assert.assertEquals(HttpStatus.SC_METHOD_NOT_ALLOWED, response.getStatusCode());
 
+        // username has special characters
+        response = rh.executePutRequest(ENDPOINT + "/internalusers/n@ag:ilum", "{\"hash\": \"123\"}", new Header[0]);
+        Assert.assertEquals(HttpStatus.SC_METHOD_NOT_ALLOWED, response.getStatusCode());
```

Comment:
Is this test case specifically related to this fix? If yes, why isn't it also `HttpStatus.SC_BAD_REQUEST`.

Reply:
Yes, we need to updated all test cases that use special characters in `username` to now fail at creation

Reply:
Agreed. Then shouldn't this `PUT` request with special characters in `username` get `HttpStatus.SC_BAD_REQUEST`, which would be consistent with the fix?

Reply:
Sounds good, updated to `HttpStatus.SC_BAD_REQUEST`.